### DISCUSSION TITLE
Telegram as SMS provider

### DISCUSSION THREAD

**Discussion by Vladimir-Va** [casdoor/casdoor#1378] [Ideas]
Hello Casdoor team.
Is possible add Telegram provider for send message for telegram bot?
I think that when implementing [this request](https://github.com/casdoor/casdoor/discussions/1358) it will be very helpful.

**Comment by forestmgy** [comment]
I'm confused about your description, why do you need telegram as sms provider? Can you tell us about your scenario?

**Reply by Vladimir-Va** [reply]
Hello i have CRM where i add Bot API то SMS provider list and send message (2FA or any other) for user who have field TelegramID. In connection with this function [Telegram as OAuth providers](https://github.com/casdoor/casdoor/discussions/1358) it will be relevant.
